### ISSUE TITLE
ChopinEarlyEditionsCollection by UniversityOfChicagoLibrary chopin.lib.uchicago.edu

### ISSUE THREAD

**Issue by Rexadev** [openzim/zim-requests#604]
*Please use the following format for a ZIM creation request (and delete unnecessary information)*

- Website URL: http://chopin.lib.uchicago.edu/
- License: **Public domain**
- Desired ZIM Title: **Chopin Sheet Music - The Chopin Early Editions collection by University of Chicago Library**
- Desired ZIM Description: **The Chopin collection at the University of Chicago Library includes over 400 first and early printed editions of musical compositions by Fr?d?ric Chopin, maintained in the Hanna Holborn Gray Special Collections Research Center. Because Chopin's works were often published concurrently in several countries with variant texts, scholars can establish a sequence of publication by comparing a range of printings. Chopin Early Editions consist of digitized images of all scores in the University of Chicago Library's Chopin collection. Users can search or browse Chopin Early Editions via a variety of data points, including titles, genres, and plate numbers.**
- Desired ZIM Icon –png (URL or attach one): https://example.com/example_icon.png
- Language (ISO 639-3): **eng**
- Is this a MediaWiki?: **no**

**Comment by Rexadev** [comment]
@RavanJAltaie updates

**Comment by RavanJAltaie** [comment]
Recipe created
https://farm.openzim.org/recipes/chopin.lib.uchicago.edu_en_all
I'll update the library link once ready

**Comment by RavanJAltaie** [comment]
Files Succeeded 
https://dev.library.kiwix.org/viewer#chopin.lib.uchicago.edu_en_all_2024-04
but one link was not working because it re-directs to another website
https://www.lib.uchicago.edu/scrc/
I've added this URL to (Include) Parameter, @benoit74 is that correct?

**Comment by Rexadev** [comment]
https://www.lib.uchicago.edu/scrc/ also has other things 
@RavanJAltaie

**Comment by benoit74** [comment]
As @Rexadev said/confirm, we do not want to include the content of https://www.lib.uchicago.edu/scrc/ inside the ZIM, so adding them in the include list **does not help at all**.

These links which are broken should in fact lead to the online resource, so that when users are online they will be able to browse this link and when they are offline they will be presented with the regular browser saying something like "sorry we cannot access this page".

The fact that it is not displaying this message is a known zimit1 issue. I suggest to tag this as "upstream" and wait for zimit2 to solve the issue.

**Comment by RavanJAltaie** [comment]
@benoit74 I've removed the link from the include parameter.
Tagged the issue as upstream.

**Comment by benoit74** [comment]
zimit2 achieved to create a pretty cool ZIM.

However there is at least two problems left to fix.

First if you go to https://dev.library.kiwix.org/viewer#chopin.lib.uchicago.edu_en_all_2024-09/chopin.lib.uchicago.edu/gsdl/cgi-bin/library%3Fe%3Dd-000-00---0chopin--00-0-0-0prompt-10---4---Document---0-1l--1-en-Zz-1---50-home---001-001-1-0utfZz-8-0%26a%3Dd%26cl%3DCL1.1%26d%3DCHOP124.1 and select another page in the **combobox**, it is not working ; navigating with prev/next works as expected ; this is an upstream bug, I will open an issue.

![Image](https://github.com/user-attachments/assets/eca2a4c2-6d6d-48a2-8de0-b707559033bc)

Second the high res images have not been fetched while they are mostly essential ; we need to update the recipe configuration to also include `http://www.lib.uchicago.edu/chopin/images/*`.

![Image](https://github.com/user-attachments/assets/9de83141-4f58-4177-8d51-27268735f22f)

@Popolechien given the upstream bug will probably take time to fix and the combobox is not essential, we already have the next / prev links, shall we simply hide the combobox with CSS so that we can publish the ZIM? @Rexadev do you think the ZIM will still be usable without this combox box

**Comment by Popolechien** [comment]
LGTM

**Comment by benoit74** [comment]
I realize that I forgot to update the recipe to include the high res image.

However, given the ZIM is already 1.25G, I'm however not sure we want to create a even bigger ZIM which could be somewhere around 5 to 10G.

Do we want to create two flavors, one without high res image (like we have currently) and one with high res image? Or do we create only one bigger ZIM? Or do we remove the high-res link and keep only the low res? (which is not awful, but might be a bit hard to read for musician in some cases).

**Comment by Popolechien** [comment]
Not a big fan of multiple flavours. This is rather niche and I don't see this kind of content being used in places where storage is an issue (meaning that folks probably won't read score off a smartphone).

**Comment by benoit74** [comment]
ZIM is ready in dev with high res images of the scores at https://dev.library.kiwix.org/#lang=&q=chopin

As expected, ZIM size grew from 1.25 GB to 8.23GB

@Popolechien  please review and request again in prod if OK (and set a periodicity, I suggest `quarterly`)

**Comment by Popolechien** [comment]
LGTM, requested in Prod, thanks. 

Seeing how Chopin has been dead 175 years now I'd assume we won't get many updates tbh.

**Comment by benoit74** [comment]
I agree we will probably not see new scores appear, but updating the ZIM to fetch website improvements (maybe) or scraper / libzim improvements (for sure), we need to request this on a regular basis. Maybe once a year is sufficient, but given the amount of workers we have and the recipe duration, I assumed quarterly was ok. I don't mind if you set this to yearly updates. But we need to update every now and then.

**Comment by Popolechien** [comment]
Well my concern is that seeing how basic and somewhat cumbersome the website is, there's a chance they would go with a full revamp that might not sit well with zimit and result with a broken zim. The current lack of QA makes it so that we probably would not be able to see it for a while, would we?

**Comment by benoit74** [comment]
No we wouldn't, indeed, you're right.

**Comment by benoit74** [comment]
Obviously, it was requested again in dev. Requested again in prod.

**Comment by benoit74** [comment]
File is ready in prod: https://library.kiwix.org/#lang=eng&q=chopin

As discussed, recipe is kept at "manual" update, since we do not expect this ZIM to need much updates.
